src/internal.c

```diff
@@ -15284,6 +15506,25 @@ int ProcessReply(WOLFSSL* ssl)
                     }
 
                     ssl->buffers.inputBuffer.idx++;
+
+                    ret = SanityCheckMsgReceived(ssl, change_cipher_hs);
```

Comment:
Should this sanity check be moved down this far?

Reply:
Previously the packet checking was being done after the sanity fails or succeeds but in an incorrect way (for DTLS). In the case of sanity check failure the result of the packet check would be returned (assuming `DUPLICATE_MSG_E` or `OUT_OF_ORDER_E` was returned by the sanity check). I unified the packet checking and moved the sanity check immediately after confirming that we have received a correct packet.